File: DSRFQ.Web/Modules/ServerTypes/Company/CompaniesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CompaniesRow } from "./CompaniesRow";

export interface CompaniesColumns {
    Id: Column<CompaniesRow>;
    Name: Column<CompaniesRow>;
    Picture: Column<CompaniesRow>;
    OrganizationId: Column<CompaniesRow>;
    InsertDate: Column<CompaniesRow>;
    InsertBy: Column<CompaniesRow>;
    UpdateDate: Column<CompaniesRow>;
    UpdateBy: Column<CompaniesRow>;
    DeleteDate: Column<CompaniesRow>;
    DeleteBy: Column<CompaniesRow>;
}

export class CompaniesColumns extends ColumnsBase<CompaniesRow> {
    static readonly columnsKey = 'Company.Companies';
    static readonly Fields = fieldsProxy<CompaniesColumns>();
}

File: DSRFQ.Web/Modules/Company/Companies/CompaniesColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Company.Columns;

[ColumnsScript("Company.Companies")]
[BasedOnRow(typeof(CompaniesRow), CheckNames = true)]
public class CompaniesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Picture { get; set; }
    public string OrganizationId { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}